fabric-server-languages-v0/src/main/java/net/fabricmc/fabric/mixin/server/languages/v0/MixinLanguage.java

```diff
@@ -0,0 +1,93 @@
+package net.fabricmc.fabric.mixin.server.languages.v0;
+
+import com.google.gson.JsonParseException;
+import net.fabricmc.fabric.impl.server.languages.v0.ServerLanguagesDuck;
+import net.fabricmc.loader.api.FabricLoader;
+import net.minecraft.client.font.TextVisitFactory;
+import net.minecraft.text.OrderedText;
+import net.minecraft.text.StringVisitable;
+import net.minecraft.text.Style;
+import net.minecraft.util.Language;
+import org.apache.logging.log4j.Logger;
+import org.spongepowered.asm.mixin.Final;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.Shadow;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+import java.util.function.BiConsumer;
+
+@Mixin(Language.class)
+public class MixinLanguage implements ServerLanguagesDuck {
+	@Shadow @Final private static Logger LOGGER;
+	@Shadow @Final private static String DEFAULT_LANGUAGE;
+	@Shadow private static volatile Language instance;
+
+	private static final Map<String, String> modLanguages = new HashMap<>();
+
+	@Shadow
+	private static void load(InputStream inputStream, BiConsumer<String, String> entryConsumer) {
+	}
+
+	private static void loadModLangFile(String modId, String langCode, BiConsumer<String, String> biConsumer) {
+		// Make sure even if somebody puts a map entry in for minecraft we still load translation mappings for en_us,
+		// as no other mappings exist in server jar.
+		if (modId.equals("minecraft")) langCode = DEFAULT_LANGUAGE;
+
+		String path = "/assets/" + modId + "/lang/" + langCode + ".json";
+
+		try (InputStream inputStream = Language.class.getResourceAsStream(path)) {
```

Comment:
This one appears wrong. You should query the mod content with the mod container instance; the language class may not be in the same jar as the mod and hence cannot get the desired json, I assume?